### DISCUSSION TITLE
Add ruff pre-commit via the pre-commit file in .git/hooks

### DISCUSSION THREAD

**Discussion by MadaraCx** [astral-sh/ruff#10974] [Q&A]
The project I'm currently working with has all it's pre-commits configured in the pre-commit file in .git/hooks (clang-tidy, clang-format). The configurations are stored in files in "." format (.clang-format, .clang-tidy). 
Is there a similar manner to implementing ruff linting at pre-commit through shell script?

**Comment by hamedrabah** [comment]
Ruff does not need a special hook-file format. A Git hook is simply an executable, so a minimal `.git/hooks/pre-commit` can run Ruff directly:

```sh
#!/bin/sh
ruff check --fix .
ruff format --check .
```

Then make it executable with `chmod +x .git/hooks/pre-commit`. Ruff will still read its normal configuration from `pyproject.toml`, `ruff.toml`, or `.ruff.toml`.

The important limitation is Git, not Ruff: files under `.git/hooks` are not version-controlled and can be replaced when the repository is recreated or another hook manager is installed. For a shared project, the official `astral-sh/ruff-pre-commit` integration is more reproducible. If your project deliberately maintains a composite hook containing clang-format and clang-tidy, adding the two Ruff commands there is valid; just keep a tracked template or installer script that copies the hook into `.git/hooks` for each developer and CI.
